File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:

        # Brute force approach:Use set to get the unique elements and then put them in the first k positions

        # Optimal approach
        i, j = 0, 1
        n = len(nums)
        for j in range(1, n):
            if nums[i] != nums[j]:
                i += 1
                nums[i] = nums[j]
        
        return i + 1

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let n = nums.length
    let i = 0

    for (let j = 1; j < n; j++){
        if (nums[i] != nums[j]){
            i += 1
            let temp = nums[i]
            nums[i] = nums[j]
            nums[j] = temp
        }
    }

    return i + 1
};